privacy-contracts/src/main/solidity/DefaultOnChainPrivacyGroupManagementContract.sol

```diff
@@ -12,54 +13,58 @@ contract DefaultOnChainPrivacyGroupManagementContract is OnChainPrivacyGroupMana
         return _version;
     }
 
-    // overrides
     function canExecute() external view returns (bool) {
         return _canExecute;
     }
 
     function lock() public {
         require(_canExecute);
+        require(tx.origin == _owner, "Origin not the owner.");
         _canExecute = false;
     }
 
     function unlock() public {
         require(!_canExecute);
+        require(tx.origin == _owner, "Origin not the owner.");
         _canExecute = true;
     }
 
-    function addParticipants(bytes32 _enclaveKey, bytes32[] memory _accounts) public returns (bool) {
+    function addParticipants(bytes32[] memory _accounts) public returns (bool) {
         require(!_canExecute);
-        if(distributionList.length == 0) {
-            addParticipant(_enclaveKey);
+        if (_owner == address(0x0)) {
+            _owner = tx.origin;
         }
```

Comment:
I don't understand why we are doing this.

Reply:
Setting the owner when the method is called the first time (just after injecting it)